controllers/tasks.js

```diff
@@ -0,0 +1,98 @@
+const Task = require('../models/Task')
+
+const addTask = (req, res) => {
+    res.render('pages/addTask')
+}
+
+const createTask = async (req, res) => {
+    try {
+        if (req.body.complete) {
+            req.body.completed=true
+        }
+        console.log(req.body, "Inside the create func")
+        await Task.create(req.body)
+        req.session.pendingMessage = "The task was created."
+        res.redirect('/tasks')
+    } catch (err) {
+        if (err.name === 'ValidationError') {
+            res.locals.message = Object.values(err.errors)
+                .map((item) => item.message)
+                .join(', ')
+        } else {
+            res.locals.message = "Something went wrong."
+        }
+        res.render('pages/addTask')
+    }
+}
+
+const deleteTask = async (req, res) => {
+    try {
+        const task = await Task.findByIdAndDelete(req.params.id)
+        req.session.pendingMessage = 'The task was deleted successfully.'
+        res.redirect('/tasks')
+    } catch (err) {
+        req.session.pendingMessage = 'Something went wrong.'
+        res.redirect('/tasks')
+    }
+}
+
+const editTask = async (req, res) => {
+    try {
+        const task = await Task.findById(req.params.id)
+        res.render('pages/editTask', { task })
+    } catch (err) {
+        req.session.pendingMessage = 'Something went wrong.'
+        res.redirect('/tasks')
+    }
+}
+
+const updateTask = async (req, res) => {
+    task = false
+    try {
+        if (req.body.complete) {         
```

Comment:
It looks like the purpose of this code is to allow the client to submit `complete` or `completed` as valid keys for the update data. What is the reason for this? Why should `complete` override any value that might already be present for `completed`. For example:

```json
{
  "complete": false,
  "completed": true
}
```

In this example, the `false` value in `complete` will override the `true` in `completed`! These types of aliases, shortcuts, or alternative namings are usually an annoying thing to support as your project grows and changes.

Assuming this behavior is something you want, and there's a reason for it, that's fine. I'd probably approach writing the code like this:

```javascript
if (req.body.completed === undefined && req.body.complete !== undefined) {
    // the android app was shipped with a bug where the app uses the name
    // "completed" instead of "complete." Using "completed" as an alias for
    // "complete" is not officially supported, and will be removed after the
    // Android team can deploy a fix
    req.body.completed = req.body.complete;
}
```

Notice that

1. We only overwrite `complete` if `complete` is already undefined
2. We add a comment with (imaginary) rationale for why we're doing this weird thing
3. We can just copy the value from `req.body.complete` into `req.body.completed`. There is no need for two if statements.

Reply:
in the database we have the field completed. i am not sure why they used complete instead of completed in the front end but its just a place holder as the value coming for the complete property is a string. And with the if statement we are trying to convert the string to boolean and pass it to the Database. So even if it comes as complete or any name the string is changed to boolean value in the backend with if condition.

Reply:
`if (req.body.complete) {         
            req.body.completed=true
        }
         else {
            req.body.completed=false
         }`